xmtp_id/src/erc1271_verifier.rs

```diff
@@ -0,0 +1,45 @@
+use ethers::types::{Address, BlockNumber, Bytes};
+use ethers::providers::{Http, Middleware, Provider};
+use std::convert::TryFrom;
+use std::sync::Arc;
+use anyhow::Error;
+use ethers::contract::abigen;
+
+const EIP1271_MAGIC_VALUE: [u8; 4] = [0x16, 0x26, 0xba, 0x7e];
+
+abigen!(
+    ERC1271,
+    "./abi/ERC1271.json",
+    derives(serde::Serialize, serde::Deserialize)
+);
+
+pub struct ERC1271Verifier {
+    pub provider: Arc<Provider<Http>>,
+}   
+
+impl ERC1271Verifier {
+    pub fn new(url: &str) -> Self {
+        let provider =  Arc::new(Provider::<Http>::try_from(url).unwrap());
+        Self { provider }
+    }
+    
+    /// Verifies an ERC-1271(https://eips.ethereum.org/EIPS/eip-1271) signature.
+    ///
+    /// # Arguments
+    ///
+    /// * `wallet_address` - Address of the ERC1271 wallet.
```

Comment:
We might want to call this `contract_address` instead, just so it doesn't cause confusion